frame/dvm/ethereum/src/lib.rs

```diff
@@ -927,63 +880,6 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
-/// The handler for interacting with The internal transaction.
-pub trait InternalTransactHandler {
-	/// Internal transaction call.
-	fn internal_transact(target: H160, input: Vec<u8>) -> DispatchResultWithPostInfo;
-	/// Read-only call to deployed evm contracts.
-	fn read_only_call(contract: H160, input: Vec<u8>) -> Result<Vec<u8>, DispatchError>;
-}
-
-impl<T: Config> InternalTransactHandler for Pallet<T> {
-	/// Execute transaction from other pallets(internal transaction)
-	/// NOTE: The difference between the rpc transaction and the internal transaction is that
-	/// The internal transactions will catch and throw evm error comes from runner to caller.
-	fn internal_transact(target: H160, input: Vec<u8>) -> DispatchResultWithPostInfo {
-		let source = T::PalletId::get().derive_ethereum_address();
```

Comment:
@hujw77 HelixDAO might have some token migration for this? The allocated part from Darwinia to Crab as xRING?

@xiaoch05

Reply:
we need this internal_transact and root_transact to migrate xRING from a pallet to a ethereum account.
see https://github.com/darwinia-network/darwinia/issues/726

Reply:
Can we do this operation on the Crab network before this pr has been upgraded to the Crab runtime? Because the internal_transact is still available in the current runtime(on-chain).

Reply:
Yes, I recommend to do that before runtime upgrade